pkg/objstore/s3/s3.go

```diff
@@ -99,22 +100,43 @@ func (conf *Config) Validate() error {
 func (conf *Config) ValidateForTests() error {
 	if conf.Endpoint == "" ||
 		conf.AccessKey == "" ||
-		conf.SecretKey == "" {
+		conf.secretKey == "" {
 		return errors.New("insufficient s3 test configuration information")
 	}
 	return nil
 }
 
 // NewBucket returns a new Bucket using the provided s3 config values.
 func NewBucket(logger log.Logger, conf *Config, reg prometheus.Registerer, component string) (*Bucket, error) {
-	var f func(string, string, string, bool) (*minio.Client, error)
-	if conf.SignatureV2 {
-		f = minio.NewV2
+	var chain []credentials.Provider
+	if conf.AccessKey != "" {
+		var signature credentials.SignatureType
```

Comment:
Minor nit: We could possibly set `signature` as V4 here and set to v2 if `conf.SignatureV2` is true. (Thus avoid not necessary `else`)